### ISSUE TITLE
panic  when resolve #someid in other file

### ISSUE THREAD

**Issue by shibormot** [santhosh-tekuri/jsonschema#18]
Hello, Santhosh Tekuri, may be I'm doing something wrong, but I got panic in Compile in this code:
```
	schema_d := `
{
    "type": "object",
    "properties": {
	"myid": { "type": "integer" },
	"otype": { "$ref": "defs.json#someid" }
    }
}
`
	defs_d := `
{
    "definitions": {
	"stt": {
	    "$schema": "http://json-schema.org/draft-07/schema#",
	    "$id": "#someid",
            "type": "object",
	    "enum": [ { "name": "stainless" }, { "name": "zinc" } ]
	}
    }
}
`
	instance_d := `
{
    "myid": 16,
    "otype": { "name": "zinc" }
}
`

	c := jsonschema.NewCompiler()
	c.Draft = jsonschema.Draft7

	schema_r := strings.NewReader(schema_d)
	c.AddResource("schema.json", schema_r)

	defs_r := strings.NewReader(defs_d)
	c.AddResource("defs.json", defs_r)

	schema, err := c.Compile("schema.json") //    <--- panic here
	if err != nil {
		log.Printf("compile: %v", err)
		return
	}

	instance_r := strings.NewReader(instance_d)
	err = schema.Validate(instance_r)
	if err != nil {
		log.Printf("validate: %v", err)
		return
	}

```

**Comment by santhosh-tekuri** [comment]
I am able to reproduce the issue. thanks for reporting.

looking into the issue

**Comment by santhosh-tekuri** [comment]
fixed with release v1.2.4

**Comment by shibormot** [comment]
Perfect! Thanks)
